sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3291,7 +3291,8 @@ pi_result piMemRelease(pi_mem Mem) {
         if (enableBufferPooling()) {
           PI_CALL(piextUSMFree(Mem->Context, Mem->getZeHandle()));
         } else {
-          ZeMemFreeHelper(Mem->Context, Mem->getZeHandle());
+          if (auto Res = ZeMemFreeHelper(Mem->Context, Mem->getZeHandle()))
```

Comment:
For all the calls to ZeMemFreeHelper, that function only returns SUCCESS (which is 0). So why check the return value for non-zero, when we know that will never happen?

Reply:
ZeMemFreeHelper can return error code as well, please take a look at this code:
```
  ZE_CALL(zeMemFree, (Context->ZeContext, Ptr));

  if (IndirectAccessTrackingEnabled)
    PI_CALL(ContextReleaseHelper(Context));
```
For example, if zeMemFree call fail, it will return error code which will be mapped to pi error code and returned.

Reply:
OK, I see.